### PR TITLE
refactor(domain): restructure FDP processing to use LDP hierarchy

### PR DESCRIPTION
**Pull request by hcvdwerf**
# **Refactor FDP processing: Use LDP hierarchy and modular structure**

## **Description**  
This refactor improves the **FairDataPointRecordProvider** by:  
- Replacing the fixed **Catalog → Dataset** structure with a **Linked Data Platform (LDP) hierarchical traversal**.  
- Separating concerns by moving `FdpRecord` and `GraphToFdpRecordMapper` into their own files.  
- Enhancing maintainability and modularity for easier future improvements.  

## **Context**  
The previous implementation **only supported a flat Catalog → Dataset structure**. However, FDP metadata can also be structured as **Study → Population → DataSeries → Dataset**.  
This update ensures that **all LDP containers** are properly traversed and processed dynamically, regardless of their hierarchy.  
This refactor also improves **code organization** for better maintainability.  

### **Major Refactors**  
- Moved `FdpRecord` to **`fdp_record.py`**  
- Moved `GraphToFdpRecordMapper` to **`graph_to_fdp_record_mapper.py`**  
- Updated `FairDataPointRecordProvider` to **import from new module structure**  
- Introduced **LDP-based traversal** instead of hardcoded Catalog → Dataset paths.  

### **Test Cases**  
✅ **Catalog-based hierarchy** (`Catalog → Dataset`)  
✅ **LDP-based hierarchy** (`Study → Population → DataSeries → Dataset`)  
✅ **Circular reference prevention** (ensuring no infinite loops)  
✅ **Edge cases** with missing relationships  

### **Test Environment**  
- Python **3.x**  
- `rdflib` for RDF parsing  
- Local FDP instance with test data  

## **Screenshots (if applicable)**  
N/A (Not a UI-related change)  

## **Additional Information**  
- This change improves **scalability**, allowing more flexible FDP structures without requiring code modifications.  
- The **refactored structure** makes it easier to extend and maintain the FDP processing logic.  

### **Code Quality**  
- [x] Code follows project style guidelines  
- [x] Code is well-commented  

### **Self-Review**  
- [x] Performed self-review and fixed typos  

### **Documentation**  
- [x] Updated relevant documentation (if needed)  

### **Testing**  
- [x] No new warnings/errors introduced  
- [x] All unit tests pass  

### **Feature Validation**  
- [x] Added tests to verify LDP traversal  
- [x] Verified expected results with different FDP structures  

---

🚀 **Ready for Review!** 🚀

## Summary by Sourcery

Refactors the FairDataPointRecordProvider to use a Linked Data Platform (LDP) hierarchical traversal for processing FDP metadata, replacing the previous fixed Catalog → Dataset structure. This change improves scalability and maintainability by allowing more flexible FDP structures without requiring code modifications.

Enhancements:
- Implements LDP-based traversal to support various FDP metadata structures, such as Study → Population → DataSeries → Dataset.
- Improves code organization by separating concerns into dedicated modules for FdpRecord and GraphToFdpRecordMapper.
- Enhances the modularity and maintainability of the FDP processing logic.

### CODE REVIEW

File: ckanext/fairdatapoint/harvesters/domain/fdp_record.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# SPDX-FileContributor: 2024 Stichting Health-RI
+#
+# SPDX-License-Identifier: AGPL-3.0-onl
+
+from rdflib import DCAT, RDF, URIRef
+
+
+class FdpRecord:
+    def __init__(self, url, graph):
+        self.graph = graph
```
Comment (PatrickDekkerHealthRI) [inline ckanext/fairdatapoint/harvesters/domain/fdp_record.py:11]: Should graph be private ? I Don't see usage outside of this graph.

**Comment by PatrickDekkerHealthRI** [comment]
I've some minor remark on this pull request.

I think it's pity that scan through fair-data-point just two collect id's and that fetch the records again for futher processing. But that a bit outside the scope of this PR.

**Review by PatrickDekkerHealthRI — APPROVED** [review]
Looks good.
